### PR TITLE
Modifications for bit-for-bit with current main in CICE

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Modifications for bit-for-bit with current main in CICE
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    This recovers bit-for-bit in CICE, will need CICE mods, still testing
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [X] Yes
    - [ ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

- Add "_old" tfrz_options to recover bit-for-bit for some cases
- Move hi_min namelist from forcing_nml to tracer_nml

### CODE REVIEW

**Comment by apcraig** [comment]
@dabail10 @eclare108213 

I know this looks odd, but it's just temporary to recover bit-for-bit in CICE.  I also need to commit some changes to the tfrz branch in CICE.  We will deprecate the "_old" options at some point, but I think we need this to preserve testing for now.  We can discuss more when we have a chance, but I'd like to fix this today or tomorrow and be able to test the heads of the branches again just to be sure.

Please merge this when you have a chance, so I can update CICE.  There may be another Icepack PRs in the next day or so if I find other issues, but I think we're OK for the moment.  Lets merge this first.

File: columnphysics/icepack_tracers.F90 [hunk 1]
```diff
@@ -1289,7 +1289,18 @@ subroutine icepack_compute_tracers (ntrcr,     trcr_depend,    &
...
                trcrn(it) = c0
-               if (it == nt_Tsfc) trcrn(it) = Tf  ! surface temperature
+               if (it == nt_Tsfc) then
+! tcraig, these old options should be deprecated
+! exist for bit-for-bit backwards compatibility in testing
+                  if (tfrz_option == "mushy_old" .or. &
+                      tfrz_option == "linear_salt_old" .or. &
+                      tfrz_option == "constant_old" .or. &
+                      tfrz_option == "minus1p8_old") then
+                     trcrn(it) = Tocnfrz  ! surface temperature
```
Comment (eclare108213) [inline columnphysics/icepack_tracers.F90:1299]: Okay, but can we fix this fairly quickly?  E.g. do the testing now to confirm that this indeed is what changes answers, then fix it and start using updated output for testing going forward?  Maybe there are a few steps along the way, but hopefully only a few.

**Review by eclare108213 — APPROVED** [review]

File: columnphysics/icepack_tracers.F90 (hunk 1 above)
Comment (apcraig) [reply columnphysics/icepack_tracers.F90:1299]: We can do that.  But my concern is that we'll start to diverge from the results on Consortium Main.  We are still bit-for-bit for all tests.  If we diverge, it's going to be hard to confirm that when we sync up the Consortium and E3SM branches, that everything is working properly and we got all the merged code correctly.  What we could also think about doing is modify Consortium Main so it gets the Tf modifications.  Or we can let them diverge.

**Comment by eclare108213** [comment]
I see.  It's important to not let them diverge.  I'd be inclined to pull over the Tf modifications, although that's a bit more work!

**Pull request merged**